src/model/table.py

```diff
@@ -55,18 +81,19 @@ def __get_all_primary_keys__(self):
 
     def get_by_primary_key(self, primary_key):
         path = self.__get_primary_key_path__(primary_key)
-        if not os.path.exists(path):
+        if path is None or not os.path.isfile(path):
             return None
         with open(path, 'r') as file:
-            return json.load(file)
+            data = json.load(file)
+        return Row(self, data)
 
     @staticmethod
     def __filter_by_query__(found_objects, query):
         if not query or str(query).isspace():
             return found_objects
         filtered_objects = []
         for object_to_compare in found_objects:
-            if Table.compare(object_to_compare, query):
+            if Table.compare(object_to_compare.data, query):
```

Comment:
Remove compare method. Move it to class Row.

so u have row.has_data(query) —> return true or false